pkg/syncer/cnsoperator/controller/cnsunregistervolume/cnsunregistervolume_controller.go

```diff
@@ -177,7 +181,198 @@ func (r *ReconcileCnsUnregisterVolume) Reconcile(ctx context.Context,
 	log.Infof("Reconciling CnsUnregisterVolume instance %q from namespace %q. timeout %q seconds",
 		instance.Name, request.Namespace, timeout)
 
-	//TODO - Add implementation logic for the reconciler
+	// 1. Perform all the necessary validations.
```

Comment:
You could hit a couple of scenarios:
1. PV deleted, PVC deleted, but CNS entry exists
2. PV PVC exists. CNS entry does not exist
3. Either of PV and PVC exists, and CNS entry may or may not exist

Some of these may not be possible in practice but can occur in theory. Does the code handle those scenarios?
My point is simply requeing on error may not be necessary, progress can be made.

Reply:
1. Is it possible that CNS metadata doesn't have PV details, and based on that can we assume that PV is deleted? PVC deleted and CNS entry existing has been handled.
2. Will enhance the code to check if error returned from CNS volume deletion is NotFound, then NOT to requeue the request.
3. I believe this is similar to 2, and can be handled in the same way.

Reply:
Address in the latest commit. And as per offline discussion, if we can't find CNS entry then we can't determine PV/PVC details and can't proceeding on deleting them. They will however get deleted during CSI full sync.  
And if only CNS entry exists but it doesn't have any PV/PVC details, then we don't operate on PV/PVC objects and just proceed with deleting CNS volume.